### PR TITLE
feat: add new Button

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
### PR Checklist

- [x] Related to designs: https://www.figma.com/file/N6TfoPI5OA7GanX6Zl8nWj/Buttons-%E2%9C%85?node-id=268%3A16
- [x] Related to JIRA ticket: [GM-121](https://codecademy.atlassian.net/browse/GM-121)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### Description

Adds five new buttons per the designs:

* `CTAButton`
* `FillButton`
* `IconButton`
* `StrokeButton`
* `TextButton`

Each uses a standard `ButtonOutline` component for the outline and a standard `ButtonInner` component for its contents.

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.4.6-alpha.88e120.0<br/>@codecademy/gamut@17.0.0-alpha.88e120.0<br/>@codecademy/styleguide@18.0.0-alpha.88e120.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f90783753ac911063ea0b46--gamut-preview.netlify.app>https://5f90783753ac911063ea0b46--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f90783753ac911063ea0b46>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.4.6-alpha.e010d4.0<br/>@codecademy/gamut@17.0.0-alpha.e010d4.0<br/>@codecademy/styleguide@18.0.0-alpha.e010d4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f907a4a8120680d9f4c3dfe--gamut-preview.netlify.app>https://5f907a4a8120680d9f4c3dfe--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f907a4a8120680d9f4c3dfe>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.976768.0<br/>@codecademy/gamut@17.0.0-alpha.976768.0<br/>@codecademy/styleguide@18.0.0-alpha.976768.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f91a8ec1b7bec00f9e518dd--gamut-preview.netlify.app>https://5f91a8ec1b7bec00f9e518dd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f91a8ec1b7bec00f9e518dd>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.3d1988.0<br/>@codecademy/gamut@17.0.0-alpha.3d1988.0<br/>@codecademy/styleguide@18.0.0-alpha.3d1988.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f92dc17294205990e798dd5--gamut-preview.netlify.app>https://5f92dc17294205990e798dd5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f92dc17294205990e798dd5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.cc57c7.0<br/>@codecademy/gamut-styles@7.1.2-alpha.cc57c7.0<br/>@codecademy/gamut@17.0.0-alpha.cc57c7.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.cc57c7.0<br/>@codecademy/styleguide@18.0.0-alpha.cc57c7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f92e2096c387fa63e8892b3--gamut-preview.netlify.app>https://5f92e2096c387fa63e8892b3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f92e2096c387fa63e8892b3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.b02f7d.0<br/>@codecademy/gamut-styles@7.1.2-alpha.b02f7d.0<br/>@codecademy/gamut@17.0.0-alpha.b02f7d.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.b02f7d.0<br/>@codecademy/styleguide@18.0.0-alpha.b02f7d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f92e523cca55ea0f4d58e5f--gamut-preview.netlify.app>https://5f92e523cca55ea0f4d58e5f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f92e523cca55ea0f4d58e5f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.965d71.0<br/>@codecademy/gamut-styles@7.1.2-alpha.965d71.0<br/>@codecademy/gamut@17.0.0-alpha.965d71.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.965d71.0<br/>@codecademy/styleguide@18.0.0-alpha.965d71.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f931bb11080491ada8cbf97--gamut-preview.netlify.app>https://5f931bb11080491ada8cbf97--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f931bb11080491ada8cbf97>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.c0f212.0<br/>@codecademy/gamut-styles@7.1.2-alpha.c0f212.0<br/>@codecademy/gamut@17.0.0-alpha.c0f212.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.c0f212.0<br/>@codecademy/styleguide@18.0.0-alpha.c0f212.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f93251c0e058d214f8b4ae7--gamut-preview.netlify.app>https://5f93251c0e058d214f8b4ae7--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f93251c0e058d214f8b4ae7>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.20b5d2.0<br/>@codecademy/gamut-styles@7.1.2-alpha.20b5d2.0<br/>@codecademy/gamut@17.0.0-alpha.20b5d2.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.20b5d2.0<br/>@codecademy/styleguide@18.0.0-alpha.20b5d2.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f93262398c2a52c3a7899ed--gamut-preview.netlify.app>https://5f93262398c2a52c3a7899ed--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f93262398c2a52c3a7899ed>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.3658cf.0<br/>@codecademy/gamut-styles@7.1.2-alpha.3658cf.0<br/>@codecademy/gamut@17.0.0-alpha.3658cf.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.3658cf.0<br/>@codecademy/styleguide@18.0.0-alpha.3658cf.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f93281410804927878cbea4--gamut-preview.netlify.app>https://5f93281410804927878cbea4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f93281410804927878cbea4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.23d137.0<br/>@codecademy/gamut-styles@7.1.2-alpha.23d137.0<br/>@codecademy/gamut@17.0.0-alpha.23d137.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.23d137.0<br/>@codecademy/styleguide@18.0.0-alpha.23d137.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f932f37f6aab73c3e0c6d7d--gamut-preview.netlify.app>https://5f932f37f6aab73c3e0c6d7d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f932f37f6aab73c3e0c6d7d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.efebfd.0<br/>@codecademy/gamut-styles@7.1.2-alpha.efebfd.0<br/>@codecademy/gamut@17.0.0-alpha.efebfd.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.efebfd.0<br/>@codecademy/styleguide@18.0.0-alpha.efebfd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f933e7972e15244f6c9a3ba--gamut-preview.netlify.app>https://5f933e7972e15244f6c9a3ba--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f933e7972e15244f6c9a3ba>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.422cb0.0<br/>@codecademy/gamut-styles@7.1.2-alpha.422cb0.0<br/>@codecademy/gamut@17.0.0-alpha.422cb0.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.422cb0.0<br/>@codecademy/styleguide@18.0.0-alpha.422cb0.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f96e72b0c018f2b22ba2498--gamut-preview.netlify.app>https://5f96e72b0c018f2b22ba2498--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f96e72b0c018f2b22ba2498>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.35504b.0<br/>@codecademy/gamut-styles@7.1.2-alpha.35504b.0<br/>@codecademy/gamut@17.0.0-alpha.35504b.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.35504b.0<br/>@codecademy/styleguide@18.0.0-alpha.35504b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f96ec86fa9519380bd73665--gamut-preview.netlify.app>https://5f96ec86fa9519380bd73665--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f96ec86fa9519380bd73665>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.493f2a.0<br/>@codecademy/gamut-styles@7.1.2-alpha.493f2a.0<br/>@codecademy/gamut@17.0.0-alpha.493f2a.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.493f2a.0<br/>@codecademy/styleguide@18.0.0-alpha.493f2a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f96ed574db5c0358198095c--gamut-preview.netlify.app>https://5f96ed574db5c0358198095c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f96ed574db5c0358198095c>Deploy Logs</a></p>

File: packages/gamut/__tests__/__snapshots__/gamut-test.ts.snap [hunk 1]
```diff
@@ -25,7 +26,9 @@ Array [
   "Container",
   "ContentContainer",
   "createPromise",
+  "CTAButton",
```
Comment (codecaaron) [inline packages/gamut/__tests__/__snapshots__/gamut-test.ts.snap:29]: Didn't we land on doing many of these new ones in `gamut-labs/experimental` and then transitioning them down the line?

File: packages/gamut-styles/src/variables/timing.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export const timing = {
+  fast: '150ms',
+};
```
Comment (codecaaron) [inline packages/gamut-styles/src/variables/timing.ts:3]: I like this.  I wonder if this should also be velocity based or something.  (though 150 would be our standard unit).

File: packages/gamut/src/Button/CTAButton.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        background: ${mode === VisualTheme.LightMode
+          ? swatches.gray[200]
+          : swatches.gray[600]};
+        box-shadow: -3px 5px 0 1px
+          ${mode === VisualTheme.LightMode
+            ? swatches.gray[600]
+            : swatches.gray[200]};
+        color: ${mode === VisualTheme.LightMode
+          ? swatches.gray[600]
+          : swatches.gray[200]};
```
Comment (codecaaron) [inline packages/gamut/src/Button/CTAButton.tsx:44]: The ternary here feels a bit hard to read.  Is there a reason we couldn't add this to `modeColorGroups`?

File: packages/gamut/src/Button/FillButton.tsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        background-color: ${swatches.hyper['400']};
+      }
+
+      ${FillButtonOuter}:disabled & {
+        background: ${mode === VisualTheme.LightMode
+          ? swatches.gray[200]
+          : swatches.gray[600]};
+        color: ${mode === VisualTheme.LightMode
+          ? swatches.gray[600]
+          : swatches.gray[200]};
```
Comment (codecaaron) [inline packages/gamut/src/Button/FillButton.tsx:31]: Ditto

File: packages/gamut/src/Button/TextButton.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import { ButtonProps, modeColorGroups } from './shared';
+
+const TextButtonInner = styled(ButtonInner)<ButtonProps>(
+  ({ mode = VisualTheme.LightMode }: ButtonProps) => {
+    const modeColors = modeColorGroups[mode];
+
+    return css`
+      border-radius: 3px;
+      color: ${modeColors.background};
+      padding: 0.75rem 1rem;
```
Comment (codecaaron) [inline packages/gamut/src/Button/TextButton.tsx:18]: Could we use spacing constants here?

**Review by codecaaron — COMMENTED** [review]
This is looking good.  Some non-blocking feedback.  But otherwise very nice 👍

File: packages/gamut/src/Button/TextButton.tsx (hunk 5 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/src/Button/TextButton.tsx:18]: @codecaaron these don't match up to the regular spacing though 😞 . `0.75rem` is between `spacing[8]` and `spacing[16]`...

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.5.0-alpha.753984.0<br/>@codecademy/gamut-styles@7.1.2-alpha.753984.0<br/>@codecademy/gamut@17.0.0-alpha.753984.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.753984.0<br/>@codecademy/styleguide@18.0.0-alpha.753984.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f99ea4aefcb980147c42d5b--gamut-preview.netlify.app>https://5f99ea4aefcb980147c42d5b--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f99ea4aefcb980147c42d5b>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.6.3-alpha.c3ee62.0<br/>@codecademy/gamut-styles@7.1.2-alpha.c3ee62.0<br/>@codecademy/gamut@17.0.0-alpha.c3ee62.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.c3ee62.0<br/>@codecademy/styleguide@18.0.0-alpha.c3ee62.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@4.6.3-alpha.4f5ef3.0<br/>@codecademy/gamut-styles@7.1.2-alpha.4f5ef3.0<br/>@codecademy/gamut@17.0.0-alpha.4f5ef3.0<br/>@codecademy/markdown-overrides@0.3.6-alpha.4f5ef3.0<br/>@codecademy/styleguide@18.0.0-alpha.4f5ef3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f99ed1a7867c00dd054df18--gamut-preview.netlify.app>https://5f99ed1a7867c00dd054df18--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f99ed1a7867c00dd054df18>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5f99ed287d6b421248de4be0--gamut-preview.netlify.app>https://5f99ed287d6b421248de4be0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/5f99ed287d6b421248de4be0>Deploy Logs</a></p>

File: packages/gamut/__tests__/__snapshots__/gamut-test.ts.snap [hunk 6]
```diff
@@ -46,6 +49,7 @@ Array [
   "Markdown",
   "MenuItem",
   "Modal",
+  "modeColorGroups",
```
Comment (JoshuaKGoldberg) [inline packages/gamut/__tests__/__snapshots__/gamut-test.ts.snap:52]: Oh, nice, I don't want to export this. Will remove.

File: packages/gamut/__tests__/__snapshots__/gamut-test.ts.snap (hunk 1 above)
Comment (JoshuaKGoldberg) [reply packages/gamut/__tests__/__snapshots__/gamut-test.ts.snap:29]: Did we? I'm a little averse to not having a core "button" component exported in some way. If we go the experimental route, we'll have only the deprecated buttons and the experimental buttons.

File: packages/gamut/src/Button/TextButton.tsx (hunk 5 above)
Comment (codecaaron) [reply packages/gamut/src/Button/TextButton.tsx:17]: I think we're good to just add any and all multiples of 4 up to some arbitrary length.

**Comment by JoshuaKGoldberg** [comment]
Dang it, merged the branch by accident...

**Comment by JoshuaKGoldberg** [comment]
Dang it, merged the branch by accident...

**Pull request merged**